crates/oxc_linter/src/rules/eslint/no_loss_of_precision.rs

```diff
@@ -0,0 +1,298 @@
+    }
+
+    fn normalize_int(num: &str) -> NormalizedNum {
+        // specially deal with 0
+        if num == "0" {
+            return NormalizedNum { magnitude: 0, coefficient: String::from("0.") };
+        }
+
+        let significant_digits = num.trim_start_matches('0').trim_end_matches('0');
+        #[allow(clippy::cast_possible_wrap)]
+        // the length of number is larger then isize is almost impossible in real-world codebase
+        let magnitude =
+            if num.starts_with('0') { num.len() as isize - 2 } else { num.len() as isize - 1 };
+        NormalizedNum { magnitude, coefficient: Self::add_decimal_point(significant_digits) }
+    }
+
+    fn normalize_float(num: &str) -> NormalizedNum {
+        let trimmed_float = match num.trim_start_matches('0') {
+            "" => "0",
+            s => s,
+        };
+
+        return trimmed_float.strip_prefix('.').map_or_else(
+            || {
+                let trimmed_float = trimmed_float.trim_end_matches('0');
+                // unwrap here will never panic, we guarantee the input contains a `.`
+                #[allow(clippy::cast_possible_wrap)]
+                let magnitude = (trimmed_float.find('.').unwrap() - 1) as isize;
+                NormalizedNum {
+                    coefficient: Self::add_decimal_point(&trimmed_float.replace('.', "")),
+                    magnitude,
+                }
+            },
+            |stripped| {
+                let decimal_digits = stripped;
+                let significant_digits = match decimal_digits.trim_start_matches('0') {
+                    "" => "0",
+                    s => s,
+                };
+
+                #[allow(clippy::cast_possible_wrap)]
+                // the length of number is larger then isize is almost impossible in real-world codebase
+                let magnitude =
+                    significant_digits.len() as isize - decimal_digits.len() as isize - 1;
+                NormalizedNum {
+                    coefficient: Self::add_decimal_point(significant_digits),
+                    magnitude,
+                }
+            },
+        );
+    }
+
+    fn normalize(num: &str) -> String {
+        if num == "inf" {
+            return String::from("inf");
+        }
+
+        let num = num.replace('E', "e");
+        let split_num = num.split('e');
+        let split_num = split_num.collect::<Vec<_>>();
+        let original_coefficient = split_num[0];
+        let normalize_num = if num.contains('.') {
+            Self::normalize_float(original_coefficient)
+        } else {
+            Self::normalize_int(original_coefficient)
+        };
+
+        let coefficient = normalize_num.coefficient;
+        // 0 is the special case
+        if coefficient == "0." {
+            return String::from("0");
+        }
+        let magnitude = if split_num.len() > 1 {
+            split_num[1].parse::<isize>().unwrap() + normalize_num.magnitude
```

Comment:
Should we make this `unwrap` safe? i.e. return if it cannot be parsed?

Reply:
> Should we make this `unwrap` safe? i.e. return if it cannot be parsed?

It will panic when exponent is extremely large, I will fix that, thanks!